src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -223,6 +221,89 @@ class RefreshIndexTests extends QueryTest with HyperspaceSuite {
     }
   }
 
+  test(
+    "Validate refresh delete action updates appended and deleted files in metadata as" +
+      "expected, when some file gets deleted and some appended to source data.") {
+    withSQLConf(
+      IndexConstants.INDEX_LINEAGE_ENABLED -> "true",
+      IndexConstants.REFRESH_DELETE_ENABLED -> "true") {
+      withIndex(indexConfig.indexName) {
+        SampleData.save(
+          spark,
+          nonPartitionedDataPath,
+          Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+        val df = spark.read.parquet(nonPartitionedDataPath)
+        hyperspace.createIndex(df, indexConfig)
+
+        // Delete one source data file.
+        deleteDataFile(nonPartitionedDataPath)
+
+        val oldFileList = fileList(nonPartitionedDataPath).toSet
+
+        // Add some new data to source.
+        import spark.implicits._
+        SampleData.testData
+          .take(3)
+          .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+          .write
+          .mode("append")
+          .parquet(nonPartitionedDataPath)
+
+        hyperspace.refreshIndex(indexConfig.indexName)
+
+        // Check if refreshed index metadata has "appendedFiles" and "deletedFiles" updated.
```

Comment:
Verify that only "appendedFiles" has value because deleted files are "refreshed". Or something like that.

Reply:
thanks, @imback82 , fixed